src/main/java/pl/joegreen/sergeants/simulator/Empty.java

```diff
@@ -0,0 +1,71 @@
+package pl.joegreen.sergeants.simulator;
+
+import java.util.Optional;
+
+class Empty extends AbstractTile {
```

Comment:
Yes this is violating the composition over inheritance rule. I will change that.
I think Empty is a good name, because it is an empty square, unless you capture it of course. What would you suggest?
Suffix it with Tile also sounds good.